docs/source/scilla-in-depth.rst

```diff
@@ -940,6 +940,29 @@ can be used to traverse all the elements of any list:
   element and the accumulator in the opposite order from the order in
   ``list_foldl``.
 
+- ``list_foldk: ('B -> 'A -> ('B -> 'B) -> 'B) -> 'B -> (List 'A) -> 'B`` :
+  Recursively process the elements in a list according to a *fold
+  description*, while keeping track of an *accumulator*.
+  ``list_foldk`` takes three arguments, which all
+  depend on the two type variables ``'A`` and ``'B``:
+
+  - The function ``c`` describing the fold step. This function takes three
+    arguments. The first argument is the current value of the
+    accumulator (of type ``'B``). The second argument is the next list
+    element to be processed (of type ``'A``). The third argument is
+    the recursive call (of type ``'B -> 'B``). The result of the
+    function is the next value of the accumulator (of type ``'B``).
+
+  - The initial value of the accumulator ``z`` (of type ``'B``).
+
+  - The list of elements to be processed (of type ``List 'A``).
+
+  In the case of a non-empty last argument with head ``x`` and tail ``xs``
```

Comment:
I imagine this paragraph might sound a bit confusing for newcomers. Let's remove it and show how to use the third argument of the combining function by example.

Reply:
Should we do the same for nat_foldk?

Reply:
yes please

Reply:
Okay I'll explain nat_foldk through nat_eq, and just cut out lines 943 to 965.